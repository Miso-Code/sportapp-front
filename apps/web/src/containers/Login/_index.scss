.login-form {
	@apply bg-white p-4 rounded-lg shadow-md flex flex-col items-center gap-4 w-full;

	min-width: 300px;
	max-width: 466px;
}
