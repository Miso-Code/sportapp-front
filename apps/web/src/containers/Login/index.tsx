import TextFieldController from 'components/Inputs/TexFieldController'
export default function LoginContainer() {
	return (
		<form>
			<TextFieldController />
		</form>
	)
}
