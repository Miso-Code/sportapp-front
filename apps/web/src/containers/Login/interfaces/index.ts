import { FormData } from 'containers/Login/utils/schema'

export interface Props {
	onHandleSubmit: (data: FormData) => void
	isDisabled?: boolean
}
