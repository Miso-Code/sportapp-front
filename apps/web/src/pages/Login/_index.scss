.login {
	@apply flex;

	min-width: 100vw;
	min-height: 100vh;

	.section-main {
		@apply flex-1 p-2 flex flex-col items-center justify-center;
	}

	.section-secondary {
		@apply w-1/2 lg:block hidden;
	}

	.container-image {
		@apply w-full h-full;
	}

	.image-login {
		@apply object-cover object-center w-full h-full;
	}
}
