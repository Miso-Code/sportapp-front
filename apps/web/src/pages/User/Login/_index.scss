.login {
	@apply flex;

	min-height: 100vh;

	.title {
		@apply font-bold mb-7 text-6xl md:text-8xl;
	}

	.section {
		&-main {
			@apply flex-1 p-2 flex flex-col items-center justify-center pb-10;
		}
	}

	.navigation {
		@apply mt-4;
	}

	.card {
		&-login {
			@apply w-full p-7 pb-4 border;

			min-width: 18.75rem;
			max-width: 29.125rem;

			&__full {
				max-width: 44.375rem;
			}
		}

		&-title {
			@apply w-full text-center mb-6;
		}
	}
}

.alert-register-container {
	@apply absolute top-0 left-0 w-full;
}
