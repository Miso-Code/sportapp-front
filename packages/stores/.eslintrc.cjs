const eslintDefaultConfig = require('@sportapp/eslint-config/library')

module.exports = {
	...eslintDefaultConfig
}
