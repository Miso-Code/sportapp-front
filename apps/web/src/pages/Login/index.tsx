import SecondarySection from '@/components/SecondarySection'
import TransitionAlert from '@/components/TransitionAlert'
import LoginContainer from '@/containers/Login'
import { FormData } from '@/containers/Login/utils/schema'
import { Button, Paper, Typography } from '@mui/material'
import { useAuthStore } from '@sportapp/stores/src/auth'
import registerImage from 'assets/images/login-wallpaper.jpg'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import './_index.scss'

export default function Login() {
	const [alert, setAlert] = useState(false)
	const navigate = useNavigate()
	const { t } = useTranslation()
	const { loading, error } = useAuthStore()
	const { login } = useAuthStore()

	const handleSubmit = async (data: FormData) => {
		const result = await login(data)

		if (result) {
			navigate('/home')
		} else {
			setAlert(true)
		}
	}

	const handleGoToRegister = () => {
		navigate('/register')
	}

	return (
		<>
			<div className='login'>
				<main className='section-main'>
					<Typography className='title' variant='h1'>
						{t('app.name')}
					</Typography>
					<Paper variant='outlined' className='card-login'>
						<Typography className='card-title' variant='h6'>
							{t('login.default')}
						</Typography>
						<LoginContainer
							isDisabled={loading}
							onHandleSubmit={handleSubmit}
						/>

						<Button
							type='button'
							fullWidth
							disabled={loading}
							onClick={handleGoToRegister}
							variant='text'
							className='navigation'
							title={t('login.button')}>
							{t('login.register.go')}
						</Button>
					</Paper>
				</main>
				<SecondarySection
					image={registerImage}
					altImage={t('login.image')}
				/>
			</div>
			<TransitionAlert
				containerClassName='alert-register-container'
				isOpen={alert}
				handleClose={setAlert}
				message={t(error ?? 'errors.register.base')}
				severity='error'
			/>
		</>
	)
}
