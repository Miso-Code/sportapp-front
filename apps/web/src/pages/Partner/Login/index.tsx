import SecondarySection from '@/components/SecondarySection'
import TransitionAlert from '@/components/TransitionAlert'
import LoginContainer from '@/containers/Login'
import { FormData } from '@/containers/Login/utils/schema'
import { Button, Paper, Typography } from '@mui/material'
import { usePartnerAuthStore } from '@sportapp/stores/src/partner/auth'
import registerImage from 'assets/images/login-partner-wallpaper.jpg'
import { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import './_index.scss'

export default function LoginPartner() {
	const [alert, setAlert] = useState(false)
	const navigate = useNavigate()
	const { t } = useTranslation()
	const { loading, error, logout } = usePartnerAuthStore()
	const { login } = usePartnerAuthStore()

	const handleSubmit = async (data: FormData) => {
		const payload = {
			email: data.email.toLowerCase(),
			password: data.password
		}
		const result = await login(payload)

		if (result) {
			navigate('/partner/home')
		} else {
			setAlert(true)
		}
	}

	const handleGoToRegister = () => {
		navigate('/partner/register')
	}

	useEffect(() => {
		logout()
	}, [logout])

	return (
		<>
			<div className='login-partner'>
				<main className='section-main'>
					<Typography className='title' variant='h1'>
						{t('app.name')}
					</Typography>
					<Paper variant='outlined' className='card-login-partner'>
						<Typography className='card-title' variant='h6'>
							{t('login.partner.title')}
						</Typography>
						<LoginContainer
							isDisabled={loading}
							onHandleSubmit={handleSubmit}
						/>

						<Button
							type='button'
							id='go-to-register'
							fullWidth
							disabled={loading}
							onClick={handleGoToRegister}
							variant='text'
							className='navigation-login-partner-register'
							title={t('login.button')}>
							{t('login.register.go')}
						</Button>
					</Paper>
				</main>
				<SecondarySection
					image={registerImage}
					altImage={t('login.image')}
				/>
			</div>
			<TransitionAlert
				containerClassName='alert-partner-login-container'
				isOpen={alert}
				handleClose={setAlert}
				message={t(error ?? 'errors.register.base')}
				severity='error'
			/>
		</>
	)
}
