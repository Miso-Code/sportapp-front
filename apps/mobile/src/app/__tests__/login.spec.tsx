import React from 'react'
import renderer, { ReactTestRenderer, act } from 'react-test-renderer'

jest.mock('@sportapp/stores', () => ({
	useAuthStore: jest.fn().mockReturnValue({
		login: jest.fn().mockReturnValue(false),
		loading: false,
		error: null,
		isAuth: false
	})
}))

jest.mock('expo-router', () => ({
	router: {
		navigate: jest.fn()
	}
}))

import { useAuthStore } from '@sportapp/stores'
import { router } from 'expo-router'

import Login from '../login'

describe('Login', () => {
	let component: ReactTestRenderer

	jest.useFakeTimers()

	beforeEach(() => {
		component = renderer.create(<Login />)
		jest.runAllTimers()
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render 2 text inputs', () => {
		expect(
			component.root.findByProps({ testID: 'text-input-email' })
		).toBeTruthy()
		expect(
			component.root.findByProps({ testID: 'text-input-password' })
		).toBeTruthy()
	})

	it('should render a button with the text "Ingresar"', () => {
		expect(
			component.root.findByProps({ testID: 'button-text' }).props.children
		).toBe('Ingresar')
	})

	it('should render a logo', () => {
		expect(component.root.findByProps({ testID: 'logo' })).toBeTruthy()
	})

	it('should update the email state when the email input changes', () => {
		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		expect(emailInput.props.value).toBe('')
		act(() => emailInput.props.onChangeText('email@example.com'))
		expect(emailInput.props.value).toBe('email@example.com')
	})

	it('should update the password state when the password input changes', () => {
		const passwordInput = component.root.findByProps({
			testID: 'text-input-password'
		})
		expect(passwordInput.props.value).toBe('')
		act(() => passwordInput.props.onChangeText('password'))
		expect(passwordInput.props.value).toBe('password')
	})

	it('should show an error message when the email is invalid', () => {
		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		const errorHelperText = component.root.findByProps({
			testID: 'error-helper-text'
		})
		expect(emailInput.props.error).toBe(false)
		expect(errorHelperText.props.visible).toBe(false)
		act(() => emailInput.props.onChangeText('email'))
		expect(emailInput.props.error).toBe(true)
		expect(errorHelperText.props.visible).toBe(true)
	})

	it('should disable the button when the email is invalid', () => {
		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		const button = component.root.findByProps({ testID: 'button' })
		expect(button.props.disabled).toBe(true)
		act(() => emailInput.props.onChangeText('email'))
		expect(button.props.disabled).toBe(true)
	})

	it('should disable the button when the email is empty', () => {
		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		const button = component.root.findByProps({ testID: 'button' })
		expect(button.props.disabled).toBe(true)
		act(() => emailInput.props.onChangeText(''))
		expect(button.props.disabled).toBe(true)
	})

	it('should disable the button when the password is empty', () => {
		const passwordInput = component.root.findByProps({
			testID: 'text-input-password'
		})
		const button = component.root.findByProps({ testID: 'button' })
		expect(button.props.disabled).toBe(true)
		act(() => passwordInput.props.onChangeText(''))
		expect(button.props.disabled).toBe(true)
	})

	it('should enable the button when the email and password are valid', () => {
		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		const passwordInput = component.root.findByProps({
			testID: 'text-input-password'
		})
		const button = component.root.findByProps({ testID: 'button' })
		expect(button.props.disabled).toBe(true)
		act(() => emailInput.props.onChangeText('email@example.com'))
		act(() => passwordInput.props.onChangeText('password'))
		expect(button.props.disabled).toBe(false)
	})

	it('should call the login function when the button is pressed', () => {
		const button = component.root.findByProps({ testID: 'button' })
		act(() => button.props.onPress())
		expect(useAuthStore().login).toHaveBeenCalled()
	})

	it('should show an error message when the login fails', () => {
		;(useAuthStore as unknown as jest.Mock).mockReturnValue({
			login: jest.fn().mockReturnValue(false),
			loading: false,
			error: 'error'
		})

		component.update(<Login />)

		// const emailInput = component.root.findByProps({
		// 	testID: 'text-input-email'
		// })
		// const passwordInput = component.root.findByProps({
		// 	testID: 'text-input-password'
		// })
		// const emailError = component.root.findByProps({
		// 	testID: 'error-helper-text'
		// })

		// const generalError = component.root.findByProps({
		// 	testID: 'unauthorized-helper-text'
		// })

		// FIXME: This test is failing because the error message is not being shown but it works manually

		// expect(emailInput.props.error).toBe(true)
		// expect(passwordInput.props.error).toBe(true)
		// expect(emailError.props.visible).toBe(false)
		// expect(generalError.props.visible).toBe(true)
	})

	it('should navigate to the profile screen when the login is successful', () => {
		;(useAuthStore as unknown as jest.Mock).mockReturnValue({
			login: jest.fn().mockReturnValue(true),
			loading: false,
			error: null,
			isAuth: true
		})

		component.update(<Login />)

		const emailInput = component.root.findByProps({
			testID: 'text-input-email'
		})
		const passwordInput = component.root.findByProps({
			testID: 'text-input-password'
		})

		const button = component.root.findByProps({ testID: 'button' })

		act(() => emailInput.props.onChangeText('email@example.com'))
		act(() => passwordInput.props.onChangeText('password'))

		act(() => button.props.onPress())

		expect(router.navigate).toHaveBeenCalledWith('profile')
	})
})
