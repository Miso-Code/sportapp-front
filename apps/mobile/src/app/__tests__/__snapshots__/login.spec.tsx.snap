// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 0,
      }
    }
  >
    <Image
      source={
        {
          "default": "mocked-file-content",
        }
      }
      style={
        {
          "backgroundColor": "grey",
          "borderRadius": 75,
          "height": 150,
          "marginBottom": 0,
          "width": 150,
        }
      }
      testID="logo"
    />
  </View>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 48,
          "fontWeight": "bold",
          "marginBottom": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    app.name
  </Text>
  <View
    style={
      {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 15,
      }
    }
  >
    <View
      style={{}}
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            false,
            {
              "backgroundColor": "rgba(255, 251, 254, 1)",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            undefined,
          ]
        }
        testID="text-input-outline"
      />
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 56,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 8,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": -0.6666666666666667,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 33,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "width": -16,
                "writingDirection": "ltr",
              }
            }
            testID="text-input-email-label-background"
          >
            form.email
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            onTextLayout={[Function]}
            style={
              {
                "color": "rgba(103, 80, 164, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-email-label-active"
          >
            form.email
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-email-label-inactive"
          >
            form.email
          </Text>
        </View>
        <TextInput
          cursorColor="rgba(103, 80, 164, 1)"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onLayout={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "margin": 0,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "minWidth": 65,
                "paddingHorizontal": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="text-input-email"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <Text
      collapsable={false}
      maxFontSizeMultiplier={1.5}
      onLayout={[Function]}
      style={
        {
          "color": "rgba(179, 38, 30, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
          "opacity": 0,
          "paddingHorizontal": 12,
          "paddingVertical": 4,
          "textAlign": "left",
          "transform": [],
          "writingDirection": "ltr",
        }
      }
      testID="error-helper-text"
    >
      validations.email
    </Text>
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            false,
            {
              "backgroundColor": "rgba(255, 251, 254, 1)",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            undefined,
          ]
        }
        testID="text-input-outline"
      />
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 56,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 750,
              "zIndex": 3,
            }
          }
        >
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 8,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "opacity": -0.6666666666666667,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 33,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "width": -16,
                "writingDirection": "ltr",
              }
            }
            testID="text-input-password-label-background"
          >
            form.password
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            onTextLayout={[Function]}
            style={
              {
                "color": "rgba(103, 80, 164, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-password-label-active"
          >
            form.password
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "maxWidth": 73,
                "opacity": 0,
                "paddingHorizontal": 16,
                "position": "absolute",
                "textAlign": "left",
                "top": 32,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
            testID="text-input-password-label-inactive"
          >
            form.password
          </Text>
        </View>
        <TextInput
          cursorColor="rgba(103, 80, 164, 1)"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onLayout={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          secureTextEntry={true}
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "margin": 0,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "minWidth": 65,
                "paddingHorizontal": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="text-input-password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 5,
          "marginTop": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(28, 27, 31, 0.12)",
            "borderColor": "transparent",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 5,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(28, 27, 31, 0.38)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              login.go
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 0,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "center",
        "transform": [],
        "writingDirection": "ltr",
      }
    }
    testID="unauthorized-helper-text"
  >
    validations.emailOrPassword
  </Text>
</View>
`;
